text/0519-2019-2020-roadmap.md

```diff
@@ -0,0 +1,90 @@
+- Start Date: 2019-07-29
+- Relevant Team(s): All teams
+- RFC PR: https://github.com/emberjs/rfcs/pull/519
+- Tracking: 
+
+# Ember 2019-2020 Roadmap
+
+## Summary
+
+This RFC sets the Ember 2019-2020 Roadmap. This year, our two priorities are:
+
+- Further reducing the size and conceptual complexity of the framework.
+- Accelerating ecosystem growth and improving ease of adoption.
+
+## Motivation
+
+This document is a distillation of multiple sources:
+
+1. The 2019 Community Survey.
+2. Community #EmberJS2019 blog posts, authored in response to our call for posts.
+3. Discussion on https://discuss.emberjs.com, Discord, Twitter, and other public venues.
+4. Deliberations among the Ember core teams.
+
+The goal of the RFC is to align the Ember community around a set of shared, achievable goals that balance the needs of existing users with the need to grow and support new users.
+
+## Detailed design
+
+Now that all of Ember Octane’s features are code complete, it’s time to turn our attention to what we’ll focus on over the next year.
+
+In this roadmap, our goal is to build on last year’s momentum and capitalize on the cutting-edge foundation we laid with Ember Octane.
+
+- **Continue simplifying Ember.** With many new APIs introduced in Octane, our focus this year should be removing features that feel out of date. The overall API surface area of Ember will shrink substantially, making it smaller, faster, and more approachable to new learners.
+- **Modernize our build system.** This year, we will put the finishing touches on Embroider and make it the default build pipeline for new applications, unlocking optimizations like tree-shaking and route-based code splitting.
+- **Better a11y by default**. We will fix the router to work out-of-the-box with assistive technologies, and give developers more tools for understanding and improving the accessibility of their Ember applications.
+- **Optimize for growth.** A feeling of momentum is critical for keeping the Ember community healthy. We will double down on evangelizing Ember Octane and making Ember approachable to a new generation of web developers.
```

Comment:
I would put Fastboot because we do not have rehydration yet, we do not have good documentation and it seems to be a forgotten project. And SSR is indispensable in the current state of web.

Reply:
I second the FastBoot goal.  I'd be interested in the number of teams that are using the library or have forked for their own needs.  It works well mostly in its current form for very large projects.  But it would be good to lay out where the community feels some good progress can be had.  I (we) have some ideas as to how to improve the ergonomics of the fastboot-app-server, but it seems like there are also ideas/concerns about it's current feature set.

Reply:
FWIW, I think [Embroider](https://github.com/embroider-build/embroider) (which is a goal) will take over the route based code splitting that FastBoot currently provides.

Reply:
@Panman8201 Fastboot doesn’t provide code splitting

Reply:
🤦‍♂ My mind was elsewhere (ember-engines).

Reply:
Ember Engines never was about code splitting. This is a confusion because engines was the unique way to reach this.

But ember engines is about isolation/separation of concerns, an antidote to many problems problems by allowing for distributed development, testing, and packaging of logically-grouped pieces of an application.